torchaudio/datasets/utils.py

```diff
@@ -76,41 +65,130 @@ def makedir_exist_ok(dirpath):
             raise
 
 
-def download_url(url, root, filename=None, md5=None):
-    """Download a file from a url and place it in root.
+def stream_url(url, start_byte=None, block_size=32 * 1024, progress_bar=True):
```

Comment:
This could be an excellent target for a multithreaded buffer function :D

Reply:
[Indeed](https://github.com/pytorch/audio/pull/323) :D